src/lib.rs

```diff
@@ -102,4 +104,27 @@ mod numinstance
     pub type Paillier = NumPaillier;
 }
 #[cfg(feature="inclnum")]
-pub use self::numinstance::*;
\ No newline at end of file
+pub use self::numinstance::*;
+
+
+/// Encryption key that may be shared publicly.
+#[derive(Debug,Clone)]
+pub struct EncryptionKey {
+    pub n: BigInteger,  // the modulus
+    nn: BigInteger, // the modulus squared
+}
+
+/// Decryption key that should be kept private.
```

Comment:
Not sure this comment is needed as this is obvious based on the struct containing only private members. What do you think?

Reply:
I see those as two unrelated things actually: access levels in code are about hiding implementation details while `private` here refers to how the key can be used in the bigger context of a cryptographic system.